extensions/sql-database-projects/src/models/project.ts

```diff
@@ -1579,12 +1675,116 @@ export class Project implements ISqlProject {
 			// Property items are expected to have simple string content
-			throw new Error(constants.invalidProjectPropertyValue(propertyName));
+			throw new Error(constants.invalidProjectPropertyValueInSqlProj(propertyName));
 		}
 
 		return firstPropertyElement.childNodes[0].nodeValue!;
 	}
 
+	/**
+	 * Retrieves all semicolon-separated values specified in the project property.
+	 *
+	 * @param propertyName Name of the project property
+	 * @returns Array of semicolon-separated values specified in the property
+	 */
+	private getCollectionProjectPropertyValue(propertyName: string): string[] {
+		const propertyValue = this.evaluateProjectPropertyValue(propertyName);
+		if (propertyValue === undefined) {
+			return [];
+		}
+
+		return propertyValue.split(';')
+			.filter(value => value.length > 0);
+	}
+
+	/**
+	 * Sets the value of the project property.
+	 *
+	 * @param propertyName Name of the project property
+	 * @param propertyValue New value of the project property
+	 */
+	private async setProjectPropertyValue(propertyName: string, propertyValue: string): Promise<void> {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		let propertyElement: Element | undefined;
+
+		// Try to find an existing property element with the requested name.
+		// There could be multiple elements in different property groups or even within the
+		// same property group (different `Condition` attribute, for example). As of now,
+		// we always choose the first one and update it.
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		for (let propertyGroupIndex = 0; propertyGroupIndex < propertyGroups.length; ++propertyGroupIndex) {
+			const propertyElements = propertyGroups[propertyGroupIndex].getElementsByTagName(propertyName);
+
+			if (propertyElements.length > 0) {
+				propertyElement = propertyElements[0];
+				break;
+			}
+		}
+
+		if (propertyElement === undefined) {
+			// If existing property element was not found, then we add a new one
+			propertyElement = this.addProjectPropertyTag(propertyName);
+		}
+
+		// Ensure property element was found or successfully added
+		if (propertyElement) {
+			if (propertyElement.childNodes.length > 0) {
+				propertyElement.replaceChild(this.projFileXmlDoc.createTextNode(propertyValue), propertyElement.childNodes[0]);
+			} else {
+				propertyElement.appendChild(this.projFileXmlDoc.createTextNode(propertyValue));
+			}
+
+			await this.serializeToProjFile(this.projFileXmlDoc);
+		}
+	}
+
+	/**
+	 * Adds an empty project property tag.
+	 *
+	 * @param propertyTag Tag to add
+	 * @returns Added HTMLElement tag
+	 */
+	private addProjectPropertyTag(propertyTag: string): HTMLElement | undefined {
+		if (this.projFileXmlDoc === undefined) {
+			return;
+		}
+
+		const propertyGroups = this.projFileXmlDoc.getElementsByTagName(constants.PropertyGroup);
+		let propertyGroup = propertyGroups.length > 0 ? propertyGroups[0] : null;
```

Comment:
use undefined instead of null since we try to use undefined instead of null throughout the code base